EightMinuteEmpireLegends/Map/Map.h

```diff
@@ -0,0 +1,109 @@
+#pragma once
+#include <string>
+#include <vector>
+
+// Class forward declarations
+
+class Map;
+struct Continent;
+class Edges;
+class Vertex;
+class Territory;
+
+// Territory class declaration
+
+class Territory
+{
+private:
+	std::string name;
+	std::string owner;	
+	int armies;		
+	std::string const c;
+public:
+	Territory(std::string name, std::string c);
+	Territory(Territory* t);
+	Territory();	
+
+	void setName(std::string name);
+	void setOwner(std::string owner);
+	void setArmies(int armies);
+	std::string getName();
+	std::string getOwner();	
+	int getArmies();
+	std::string getContinent();
+	std::string toString();
+};
+
+// Vertex class declaration
+
+class Vertex
+{
+private:
+	std::string id;
+	Territory* const t;
+public:
+	Vertex(Territory* t, std::string id);
+	Vertex();
+
+	std::string getId();
+	Territory* getTerritory();
+	bool compare(Vertex* v);
+};
+
+// Edge class declaration
+
+class Edge
+{
+private:
+	std::string id;
+	Vertex* v1;
+	Vertex* v2;
+public:
+	Edge(Vertex* v1, Vertex* v2, std::string id);
+	Edge();
+
+	std::string getId();
+	std::vector<Vertex*> getEndpoints();	
+};
+
+// Continent struct declaration
+
+struct Continent
+{
+	std::string name;
+};
+
+// Map class declaration
+
+class Map
+{
+private:
+	std::string name;
```

Comment:
please add a 'Vertex* startingRegion' ptr in the Map, so it has information about the starting region.

Reply:
Done.